### PR TITLE
Tabular: Added time_limit=None warning

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*
#1096 

*Description of changes:*

- Added warning when time_limit=None and data is large.
- Hid NumExpr log statement caused by pandas.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1098-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1098/1/index.html

**Comment by szha** [comment]
Job PR-1098-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1098/2/index.html

File: core/src/autogluon/core/utils/utils.py [hunk 1]
```diff
@@ -735,6 +736,22 @@ def _get_safe_fi_batch_count(X, num_features, X_transformed=None, max_memory_rat
...
+        previous_log_level = root_logger.getEffectiveLevel()
+        try:
+            root_logger.setLevel(max(30, previous_log_level))
+            return func(*args, **kwargs)
+        finally:
+            root_logger.setLevel(previous_log_level)
+    return inner
+
+
+# suspect_logging to hide the Pandas log of NumExpr initialization
```
Comment (jwmueller) [inline core/src/autogluon/core/utils/utils.py:753]: suspend not suspect

File: tabular/src/autogluon/tabular/predictor/predictor.py [hunk 2]
```diff
@@ -690,6 +690,13 @@ def fit(self,
                 ag_args_ensemble = {}
             ag_args_ensemble['save_bag_folds'] = kwargs['_save_bag_folds']
 
+        if time_limit is None:
+            mb_mem_usage_train_data = get_approximate_df_mem_usage(train_data, sample_ratio=0.2).sum() / 1e6
+            num_rows_train = len(train_data)
+            if mb_mem_usage_train_data >= 50 or num_rows_train >= 100000:
+                logger.log(20, f'Warning: Training may take a very long time because `time_limit` was not specified and `train_data` is large ({num_rows_train} samples, {round(mb_mem_usage_train_data, 2)} MB).')
+                logger.log(20, f'\tConsider setting `time_limit` to ensure training finishes within an expected duration or experiment with a small portion of `train_data` to identify an ideal `hyperparameters` configuration.')
```
Comment (jwmueller) [inline tabular/src/autogluon/tabular/predictor/predictor.py:698]: might also mention `presets` in addition to  `hyperparameters`

**Review by jwmueller — APPROVED** [review]
If text is present, may also want to later print that: text fields will be responsible for long training-time if user didn't specify time_limit

File: tabular/src/autogluon/tabular/predictor/predictor.py (hunk 2 above)
Comment (Innixma) [reply tabular/src/autogluon/tabular/predictor/predictor.py]: added

File: core/src/autogluon/core/utils/utils.py (hunk 1 above)
Comment (Innixma) [reply core/src/autogluon/core/utils/utils.py]: fixed

**Comment by Innixma** [comment]
> If text is present, may also want to later print that: text fields will be responsible for long training-time if user didn't specify time_limit

This is reasonable, but should hopefully be tied the large data size via memory usage (text takes up lots of memory). If we see cases where the first warning is not triggered but it still takes a long time due to text, we can add a further warning once text is identified.

**Comment by szha** [comment]
Job PR-1098-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1098/3/index.html

**Pull request merged**